File: src/Leibniz.Api/Posts/Domain/Post.cs
namespace Leibniz.Api.Posts.Domain;
public record Post : AuditableEntity
{
    [Key]
    public long PostId { get; set; }
    [MaxLength(255)]
    public string? Title { get; set; }
    public string? Content { get; set; }
    [MaxLength(255)]
    public string? Author { get; set; }
    public long? BookId { get; set; }
    public int? Page { get; set; }
    [MaxLength(500)]
    public string? Reference { get; set; }
}

File: src/Leibniz.Client/src/app/posts/domain/post.ts
import { RefItem } from '../../common/domain/ref-item';

export interface Post {
  postId: number;
  title: string;
  content: string;
  author: string;
  bookId: number | null;
  bookName?: string | null;
  page: number | null;
  reference: string | null;
  refs: RefItem[] | null;
  imageFileName?: string | null;
}
